Components: Update overflow on select dropdown component

jeremeylduvall opened the pull request:
This is a fix for two issues. First, with #9532, I added `max-width: 100%;` to `.select-dropdown__container`. Second, for #1742, I fixed that by doing a few things:
- Adding `overflow: hidden;` to `.select-dropdown__item-text`
- Adding a right side margin identical to the left since the item is absolutely positioned.
- Removing `text-overflow: ellipsis;` from `.select-dropdown__item`. With this set of changes, it was creating duplicate ellipses because we had it on `.select-dropdown__item` and `.select-dropdown__item-text`. You can see a bit of this here within the word "Template" using the Orvis theme:

<img width="340" alt="screen shot 2017-01-06 at 7 12 21 am" src="https://cloud.githubusercontent.com/assets/7240478/21720296/7fd2c16e-d3df-11e6-81ed-68a0401718b6.png">

## To test
1. Load up this branch.
2. Activate a theme with a long page template title (I tested with Twenty Twelve) on your site.
3. Open up a new page. Check out the left-hand page template selector to notice the changes. Twenty Twelve has a page template called "Full-width Page Template, No Sidebar" that demonstrates the issues well.

Some other places you can test to make sure this has no adverse effects:
- When adding a media gallery to a post or page, the select dropdown is used for the Layout and Link To fields.
- DevDocs: http://calypso.localhost:3000/devdocs/design/select-dropdown

## Screenshots

<img width="492" alt="fix 1" src="https://cloud.githubusercontent.com/assets/7240478/21720403/1d26a3c2-d3e0-11e6-9441-7ae1c7627ced.png">

<img width="445" alt="fix 2" src="https://cloud.githubusercontent.com/assets/7240478/21720405/203784be-d3e0-11e6-98aa-39316e241fbc.png">

<img width="471" alt="fix 3" src="https://cloud.githubusercontent.com/assets/7240478/21720409/254f59e0-d3e0-11e6-8481-6c8d18762882.png">

jeremeylduvall commented:
This is labeled `In Progress` because I noticed a slight issue with the Dev Docs example when the `text-overflow: ellipsis;` is removed from `.select-dropdown__item`:

<img width="303" alt="screen shot 2017-01-06 at 7 27 39 am" src="https://cloud.githubusercontent.com/assets/7240478/21720682/b609d194-d3e1-11e6-9f4e-b21fbf87dc7d.png">

Investigating.

jeremeylduvall commented:
This proved to be a bit tricky. In order to fix the duplicate ellipses present in the Editor, I added a specific class to that dropdown component (`.editor-page-templates__select-dropdown`) since this is the only place I could find the issue. Since I'm not removing `text-overflow: ellipsis`from the overall component, the Dev Docs example is no longer broken.

Same steps to test apply as above.

**Updated screenshots**

No duplicate ellipses

<img width="558" alt="screen shot 2017-01-10 at 6 43 52 am" src="https://cloud.githubusercontent.com/assets/7240478/21808360/35192824-d700-11e6-8168-47d314f7677b.png">

DevDocs example

<img width="393" alt="screen shot 2017-01-10 at 6 44 37 am" src="https://cloud.githubusercontent.com/assets/7240478/21808375/468c0cd4-d700-11e6-8fea-4d5e3d9a458f.png">

aduth commented:
Why do we have `text-overflow: ellipsis` on both `.select-dropdown__item` and `.select-dropdown__item-text`?

jeremeylduvall commented:
@aduth Great question. I tinkered with this a bit, and I think I have a fix that works on the component now and doesn't rely on any CSS within `editor-page-templates`. This makes much more sense.

To recap:

I added `max-width: 100%;` to `.select-dropdown__container` to fix the issue of the dropdown spilling out of the editor sidebar.

The duplicate `text-overflow: ellipsis;` on both `.select-dropdown__item` and `.select-dropdown__item-text` weren't necessary. I removed it from the item and kept it on the text. In order to get the overflow to kick in, I set the display to `inline-block` and a `max-width: 100%;`.

<img width="248" alt="screen shot 2017-01-12 at 6 49 02 am" src="https://cloud.githubusercontent.com/assets/7240478/21892302/6131d584-d893-11e6-9de5-10fcda4d8251.png">
<img width="260" alt="screen shot 2017-01-12 at 6 49 10 am" src="https://cloud.githubusercontent.com/assets/7240478/21892303/6132bb34-d893-11e6-86a9-69786f6dbd3f.png">

folletto commented:
Design wise seems good, the only thing I'd mention here is that Calypso doesn't use "..." ellipses, but fades out the text to the right, so if possible I'd try to do that. I understand however it's more complicated given the discussion above, so I'd be fine in pushing this in first as-is. :)
